pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -232,25 +229,44 @@ func (ku *KubeUtil) GetLocalPodList(ctx context.Context) ([]*Pod, error) {
 	// cache the podList to reduce pressure on the kubelet
 	cache.Cache.Set(podListCacheKey, pods, ku.podListCacheDuration)
 
+	return &pods, nil
+}
+
+// GetLocalPodList returns the list of pods running on the node.
+// If kubernetes_pod_expiration_duration is set, old exited pods
+// will be filtered out to keep the podlist size down: see json.go
+func (ku *KubeUtil) GetLocalPodList(ctx context.Context) ([]*Pod, error) {
```

Comment:
I'm a bit confused on how having two `(ku *KubeUtil) GetLocalPodList(ctx context.Context)` actually works 🤔

Reply:
For the purposes of the check, we need to know the number of pods that were expired from the list when we parse in the json. That is being stored on the `PodList` object, as a metadata field. Hence, `GetLocalPodListWithMetadata`.

But, this field only makes sense to this check, and so it does not make sense to update every use case of `GetLocalPodList` to do a nil check before then calling `PodList.Items`, because that is not a clean interface to interact with. Hence the `GetLocalPodList` implementation here.

But, under the hood, both of these functions are doing the same thing, it is just that one is returning the `PodList` object, and the other is returning the underlying `Pod` slice that is attached to `PodList.Items`. So, the shared logic has been pulled out and is being used by both by calling `getLocalPodList`.

I am not sure if this answers your question or not, please let me know!